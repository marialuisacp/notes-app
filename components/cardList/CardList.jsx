import React from 'react';
import StyledCardList from './styles/CardList.styles';
import CardListHeader from '../cardListHeader/CardListHeader';
import Card from '../card/Card';
import PropTypes from 'prop-types';

const CardList = ({ list, title }) => (
  <StyledCardList>
    <CardListHeader title={title}></CardListHeader>
    <div>
      {list && list.map((item, key) => <Card text={item.text} key={key}></Card>)}
    </div>
  </StyledCardList>
);

CardList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  title: PropTypes.string.isRequired
};

export default CardList;