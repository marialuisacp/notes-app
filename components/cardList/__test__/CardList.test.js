import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import CardList from '../CardList';

const defaultProps = {
  title: 'List 1',
  list: [
    { text: 'Task 1', labels: [], icons: [] },
    { text: 'Task 2', labels: [], icons: [] },
    { text: 'Task 3', labels: [], icons: [] },
  ]
};
test('should render component CardList', () => {
  const { fragment } = render(<CardList {...defaultProps}></CardList>);
  expect(fragment).toMatchSnapshot();
});

test('should CardList component with empty list', () => {
  const { fragment } = render(<CardList title='Title' list={[]}></CardList>);
  expect(fragment).toMatchSnapshot();
});

test('should have title text', () => {
  render(<CardList {...defaultProps}></CardList>);
  expect(screen.getByText(defaultProps.title)).toBeInTheDocument();
});


