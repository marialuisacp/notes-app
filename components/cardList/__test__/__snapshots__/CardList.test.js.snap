// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CardList component with empty list 1`] = `undefined`;

exports[`should render component CardList 1`] = `undefined`;
